modules/extensions-spring-support/src/main/java/org/mule/module/extension/internal/config/BaseExtensionBeanDefinitionParser.java

```diff
@@ -14,16 +14,39 @@
 import org.springframework.beans.factory.xml.ParserContext;
 import org.w3c.dom.Element;
 
+/**
+ * Base class for {@link BeanDefinitionParser}s that process elements
+ * of components developed using the Extensions API
+ *
+ * @since 3.7.0
+ */
 abstract class BaseExtensionBeanDefinitionParser implements BeanDefinitionParser
 {
 
+    /**
+     * The type of the instance to be generated
+     */
     private final Class<?> type;
 
+    /**
+     * Constructs a new instance which will generate instances of {@code type}
```

Comment:
I'll change it, but since this comment is documenting a Constructor, I think the difference is superflous